### ISSUE TITLE
Forcing one amp-ad to be the master

### ISSUE THREAD

**Issue by jdelhommeau** [ampproject/amphtml#1524]
Hi Guys,

I am currently working on the amp adaptor for appnexus adserver.
We want to support one ad call for every ad placement on the page, which require communicating among the different amp-ad tags on the page.

The idea is that the master iframe will make the ad call for every placements on the page. Then every slave iframe will ask the master iframe to give them the ad code to render.
It works fine, however, as it seems the master iframe change every time we refresh the page, we need to pass the configuration for all placements in every amp-ad tag, rather than only passing it to the master (as we don't know in advance which one will be the master).

Is there a way to either force or know in advance which amp-ad will be the master?

To give you a better idea of the problem, here is the code to setup two ads:

```

<!doctype html>
<html ⚡>
  <head>
    <meta charset="utf-8">
    <link rel="canonical" href="hello-world.html" >
    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
    <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
    <script async src="https://cdn.ampproject.org/v0.js"></script>
  </head>
<body>
  <amp-ad width=300 height=250
      type="appnexusast"
	  data-target="apn_ad_40954389058"
	  json='{"pageOpts":{"member": 958}, "adUnits": [{"member": 958,"invCode": "ast_guaranteed_prios_1","sizes": [300,250],"targetId": "apn_ad_40954389058"}, {"member": 958, "invCode": "ast_guaranteed_prios_1","sizes": [160,600],"targetId":"apn_ad_5675675648"}]}'>
  </amp-ad>
  <amp-ad width=160 height=600
      type="appnexusast"
	  data-target="apn_ad_5675675648"
	  json='{"pageOpts":{"member": 958}, "adUnits": [{"member": 958,"invCode": "ast_guaranteed_prios_1","sizes": [300,250],"targetId": "apn_ad_40954389058"}, {"member": 958, "invCode": "ast_guaranteed_prios_1","sizes": [160,600],"targetId":"apn_ad_5675675648"}]}'>
  </amp-ad>

</body>
</html>
```

What we would like to have is this:

```

<!doctype html>
<html ⚡>
  <head>
    <meta charset="utf-8">
    <link rel="canonical" href="hello-world.html" >
    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
    <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
    <script async src="https://cdn.ampproject.org/v0.js"></script>
  </head>
<body>
  <amp-ad width=300 height=250
      type="appnexusast"
	  data-target="apn_ad_40954389058"
	  json='{"pageOpts":{"member": 958}, "adUnits": [{"member": 958,"invCode": "ast_guaranteed_prios_1","sizes": [300,250],"targetId": "apn_ad_40954389058"}, {"member": 958, "invCode": "ast_guaranteed_prios_1","sizes": [160,600],"targetId":"apn_ad_5675675648"}]}'>
  </amp-ad>
  <amp-ad width=160 height=600
      type="appnexusast"
	  data-target="apn_ad_5675675648">
  </amp-ad>

</body>
</html>
```

Thank you for your help.

Julien

**Comment by cramforce** [comment]
It's not possible to know in advance which one will be master. Since in
your example only 1 of the 2 ads might load on a given page, the designated
master might never load (imagine they are far apart, one on top, one at the
bottom. You scroll down and hit reload, only the bottom ad loads).

Master selection is completely random (driven by network speed). Whichever
ad loads first, wins.

**Comment by jdelhommeau** [comment]
Thank you @cramforce , that makes sense.
In that case, is there a way to make a third element, containing the configuration, always load, ideally first, and have the two amp-ad tags only display the ad (not making any call) whenever they need it?

**Comment by cramforce** [comment]
There is currently no way to uncouple configuration from placement on the
page. I think that would be an interesting enhancement, but it would take
in the order of 4-5 weeks to fully land (due to validation changes, etc.).

**Comment by jasti** [comment]
@cramforce  Can @jdelhommeau use the option to check if the current the current adslot is the master - something similar to https://github.com/ampproject/amphtml/blob/master/3p/twitter.js#L30 ?

I can put out an ads example if it is easier.

**Comment by cramforce** [comment]
@jasti I made you #1815, to make the examples much, much simpler. Would be great to get an ads example!

**Comment by jasti** [comment]
Awesome! Waiting on merge.

**Comment by jridgewell** [comment]
Closed by #1815.
